### ISSUE TITLE
Disable icon tooltips

### ISSUE THREAD

**Issue by fabpico** [carbon-design-system/carbon#8805]
### Summary
`"carbon-components-react": "^7.34.0"`.

Icon tooltips are currently forced. In my case, I don't want any auto-generated icon tooltips.

* When I use `HeaderGlobalAction`, the `aria-label` is required, which prerenders a tooltip element with class `bx--assistive-text`
* The `TextInput.PasswordInput` also renders by default an `<span class="bx--assistive-text">Show password</span>` element. (This also sporadically generates a confusing side effect which pushes the layout out of borders when using chrome DevTools in mobile mode, even if the tooltip is hidden. By deleting the assistive text element form the DOM, the layout normalizes.)
* There may be other components that renders a tooltip

### Justification

At the moment, I hide them with css, which I dont like. The selectors are somtimes laborious to pick because they result like:
```
.bx--btn.bx--btn--icon-only.bx--tooltip__trigger:hover .bx--assistive-text,
.bx--btn.bx--btn--icon-only.bx--tooltip__trigger:focus .bx--assistive-text`,
.bx--btn.bx--btn--icon-only.bx--tooltip__trigger::before,
etc.
```
I want to avoid `!important;` solutions.

On the other hand, I think anyone should be able to decide when to have a tooltip, instead of being forced to have them.

### "Must have" functionality

I would like to have an option to prevent rendering of icon tooltips.

**Comment by tw15egan** [comment]
I believe the tooltips are an a11y requirement for icon-only elements, so I'm not sure if they should be allowed to be disabled. Thoughts @dakahn?

**Comment by dakahn** [comment]
It's not explicitly an accessibility requirement. In our system we don't allow icon only buttons to disable their tooltips. A case could be made for it (close buttons on modals for example don't get tooltips on focus).

**Comment by aagonzales** [comment]
I would say the use-case for disabling the tooltip on an icon only button is an edge case and one we don't want to encourage sense it aids in accessibility. I do believe we have set up a feature to allow developers add a delay for when the tooltip appears which could help in certain situations.

**Comment by thyhmdo** [comment]
We will need to define:
- When tooltip can be turned off and when we should keep them turned on?

**Comment by jnm2377** [comment]
Closing since we're not pursing. We can re-open if that decision changes in the future. :)
